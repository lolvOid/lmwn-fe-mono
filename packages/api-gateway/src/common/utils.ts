/**
 * A function that wraps the input function to handle asynchronous operations and errors.
 *
 * @param {Function} func - the input function to be wrapped
 * @return {Function} a function that handles asynchronous operations and errors
 */
export function asyncHandler(func: Function) {
    return (req: any, res: any, next: any) => Promise.resolve(func(req, res, next)).catch(next);
}