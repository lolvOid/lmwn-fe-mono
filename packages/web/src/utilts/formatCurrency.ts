export const getThaiBaht = (value: number | string, digit?: number) => {
    return new Intl.NumberFormat('th-TH', { style: 'currency', currency: 'THB', maximumFractionDigits: digit || 0 }).format(parseFloat(String(value)));
}