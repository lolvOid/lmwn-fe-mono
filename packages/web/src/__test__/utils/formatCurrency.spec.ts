import { getThaiBaht } from '@/utils/formatCurrency';

describe('getThaiBaht', () => {
    it('Should format the value to Thai Baht currency without digits', () => {
        const formattedValue = getThaiBaht(1000);
        expect(formattedValue).toMatchSnapshot();
    });

    it('Should format the value to Thai Baht currency with specified digits', () => {
        const formattedValue = getThaiBaht(12345.678, 2);
        expect(formattedValue).toMatchSnapshot();
    });

    it('Should be string input for value', () => {
        const formattedValue = getThaiBaht('5000');
        expect(formattedValue).toMatchSnapshot();
    });

    it('Should be invalid input', () => {
        const formattedValue = getThaiBaht('invalid');
        expect(formattedValue).toMatchSnapshot();
    });

    it('Should be undefined digit', () => {
        const formattedValue = getThaiBaht(9876.54321);
        expect(formattedValue).toMatchSnapshot();
    });
});
