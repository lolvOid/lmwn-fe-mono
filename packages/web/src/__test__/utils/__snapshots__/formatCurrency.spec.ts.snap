// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getThaiBaht Should be invalid input 1`] = `"฿NaN"`;

exports[`getThaiBaht Should be string input for value 1`] = `"฿5,000"`;

exports[`getThaiBaht Should be undefined digit 1`] = `"฿9,877"`;

exports[`getThaiBaht Should format the value to Thai Baht currency with specified digits 1`] = `"฿12,345.68"`;

exports[`getThaiBaht Should format the value to Thai Baht currency without digits 1`] = `"฿1,000"`;
